ietf/meeting/views.py

```diff
@@ -2516,6 +2527,65 @@ def add_session_drafts(request, session_id, num):
                     'form': form,
                   })
 
+def bluesheet_data(session):
+    def affiliation(meeting, person):
+        # from OidcExtraScopeClaims.scope_registration()
+        email_list = person.email_set.values_list('address')
+        q = Q(person=person, meeting=meeting) | Q(email__in=email_list, meeting=meeting)
+        regs = MeetingRegistration.objects.filter(q).distinct()
+        return ([reg.affiliation for reg in regs if reg.affiliation] or [''])[0]
+
+    attendance = Attended.objects.filter(session=session)
+    meeting = session.meeting
+    return [{'name':attended.person.plain_name(), 'affiliation':affiliation(meeting, attended.person)} for attended in attendance]
+
+def session_attendance(request, session_id, num):
+    # num is redundant, but we're dragging it along as an artifact of where we are in the current URL structure
+    session = get_object_or_404(Session, pk=session_id)
+    if session.meeting.type_id != 'ietf' or session.meeting.proceedings_final:
+        bluesheets = session.sessionpresentation_set.filter(document__type_id='bluesheets')
+        if bluesheets:
+            bluesheet = bluesheets[0].document
+            return redirect(bluesheet.get_href(session.meeting))
+        else:
+            raise Http404('Bluesheets not found')
+
+    cor_cut_off_date = session.meeting.get_submission_correction_date()
+    today_utc = date_today(datetime.timezone.utc)
+    try:
+        person = request.user.person
+        was_there = Attended.objects.filter(session=session, person=person).exists()
+        can_add = today_utc <= cor_cut_off_date and MeetingRegistration.objects.filter(meeting=session.meeting, person=person).exists() and Attended.objects.filter(session=session).exists() and not was_there
+    except AttributeError:
+        was_there = can_add = False
```

Comment:
I think something like
`if user.is_authenticated and user.person is not None:`
would be a lot better